File: license/php/roster.php
<?php

include_once "licenseA.php";
include_once "licenseB.php";
include_once "licenseC.php";
include_once "student.php";

// Simular los datos del estudiante
$studentName = "John";
$lastName = "Doe";
$identification = "123456789";
$direction = "123 Main Street";
$phone = "555-1234";
$rh = "O+";

// Simular la selección de licencia
$choice = 1; // Cambia este valor a 1, 2 o 3 para probar diferentes licencias

// Crear la licencia seleccionada
$selectedLicense = null;
if ($choice === 1) {
    $selectedLicense = new LicenseA();
} elseif ($choice === 2) {
    $selectedLicense = new LicenseB();
} elseif ($choice === 3) {
    $selectedLicense = new LicenseC();
} else {
    echo "Invalid choice. Exiting program.\n";
    exit;
}

// Crear el objeto Student con los datos ingresados
$student = new Student($studentName, $lastName, $identification, $direction, $phone, $rh, $selectedLicense);

// Mostrar los datos del estudiante registrado
echo "\n--- Student Registered ---\n";
echo "Name: " . $student->getName() . "\n";
echo "Last Name: " . $student->getLastName() . "\n";
echo "Identification: " . $student->getIdentification() . "\n";
echo "Direction: " . $student->getDirection() . "\n";
echo "Phone: " . $student->getPhone() . "\n";
echo "RH: " . $student->getRh() . "\n";
echo "License: " . $selectedLicense->getName() . "\n";
echo "Duration: " . $selectedLicense->getDuration() . "\n";
echo "Cost: $" . $selectedLicense->getCost() . "\n";

?>

File: license - copia/java/roster.java
import java.util.Scanner;

public class roster {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {
            // Solicitar los datos del estudiante
            System.out.print("Enter the student's name: ");
            String studentName = sc.nextLine();

            System.out.print("Enter the student's last name: ");
            String lastName = sc.nextLine();

            System.out.print("Enter the student's identification: ");
            String identification = sc.nextLine();

            System.out.print("Enter the student's direction: ");
            String direction = sc.nextLine();

            System.out.print("Enter the student's phone: ");
            String phone = sc.nextLine();

            System.out.print("Enter the student's RH: ");
            String rh = sc.nextLine();

            // Mostrar las opciones de licencias
            System.out.println("Select a license:");
            System.out.println("1. License A - $1,000,000 COP");
            System.out.println("2. License B - $1,500,000 COP");
            System.out.println("3. License C - $2,000,000 COP");

            System.out.print("Enter your choice (1, 2, or 3): ");
            int choice = sc.nextInt();

            // Crear la licencia seleccionada
            license selectedLicense;
            selectedLicense = switch (choice) {
                case 1 -> new licenseA();
                case 2 -> new licenseB();
                case 3 -> new licenseC();
                default -> {
                    System.out.println("Invalid choice. Exiting program.");
                    yield null;
                }
            };

            if (selectedLicense == null) {
                return;
            }

            // Crear el objeto estudiante
            student student = new student(studentName, lastName,
                    identification, direction, phone, rh, selectedLicense);

            // Mostrar los datos del estudiante registrado
            System.out.println("\n--- Estudiante Registrado ---");
            System.out.println("Nombre: " + student.getName());
            System.out.println("Apellido: " + student.getLastName());
            System.out.println("Identificación: " + student.getIdentification());
            System.out.println("Dirección: " + student.getDirection());
            System.out.println("Teléfono: " + student.getPhone());
            System.out.println("RH: " + student.getRh());
            System.out.println("Licencia: " + selectedLicense.getName());
            System.out.println("Duración: " + selectedLicense.getDuration());
            System.out.println("Costo: $" + selectedLicense.getCost());
        }
    }
}